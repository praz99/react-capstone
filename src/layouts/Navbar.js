import React from 'react';
import '../styles/Navbar.css';

const Navbar = () => (
  <div className="navbar-container">
    <div className="banner">
      <a href="/" data-testid="navbar-heading">The CockTails</a>
    </div>
    <div className="slogan">
      <span>Find your drink...</span>
      <span>Learn to make</span>
      <div>Enjoy!</div>
    </div>
  </div>
);

export default Navbar;
