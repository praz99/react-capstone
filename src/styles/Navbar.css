.navbar-container {
  width: 30%;
}

.banner a {
  font-size: 3rem;
  color: #45cccc;
  letter-spacing: 1px;
  text-decoration: none;
}

.slogan {
  font-size: 10px;
  text-align: right;
  width: 80%;
  color: #6b0646;
}
