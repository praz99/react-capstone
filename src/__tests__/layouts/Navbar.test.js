import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Navbar from '../../layouts/Navbar';

afterEach(cleanup);

it('renders Navbar component correctly', () => {
  const nav = renderer.create(<Navbar />).toJSON();
  expect(nav).toMatchSnapshot();
});

it('should display the heading', () => {
  const { getByTestId } = render(<Navbar />);
  expect(getByTestId('navbar-heading')).toHaveTextContent('The CockTails');
});

it('should display link for home page', () => {
  render(<Navbar />);
  const homeAnchorNode = screen.getByText('The CockTails');
  expect(homeAnchorNode).toBeInTheDocument();
});
