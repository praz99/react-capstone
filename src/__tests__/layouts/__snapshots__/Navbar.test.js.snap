// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Navbar component correctly 1`] = `
<div
  className="navbar-container"
>
  <div
    className="banner"
  >
    <a
      data-testid="navbar-heading"
      href="/"
    >
      The CockTails
    </a>
  </div>
  <div
    className="slogan"
  >
    <span>
      Find your drink...
    </span>
    <span>
      Learn to make
    </span>
    <div>
      Enjoy!
    </div>
  </div>
</div>
`;
